src/controller/tests/data_model/TestRead.cpp

```diff
@@ -1655,15 +1679,17 @@ void TestReadInteraction::TestReadHandler_KillOverQuotaSubscriptions(nlTestSuite
     }
     readCallback.ClearCounters();
 
-    ctx.DrainAndServiceIO();
-    // Ensure the global dirty set is cleared.
-    NL_TEST_ASSERT(apSuite, app::InteractionModelEngine::GetInstance()->GetReportingEngine().GetGlobalDirtySetSize() == 0);
+    ctx.GetIOContext().DriveIOUntil(System::Clock::Seconds16(60), [&]() {
```

Comment:
What's the purpose of this block of code? Why are we waiting to get all the attributes across all the subs?

Reply:
To validating we have evicted the correct subscription.

Added a comment for this.